workflow/controller/controller.go

```diff
@@ -136,6 +137,7 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, podWorkers in
 	}
 
 	wfc.wfInformer = util.NewWorkflowInformer(wfc.restConfig, wfc.Config.Namespace, workflowResyncPeriod, wfc.tweakWorkflowlist)
+	wfc.throttler = NewThrottler(wfc.Config.Parallelism, wfc.wfQueue)
```

Comment:
The workflow controller is supposed to notice changes to the configmap and have the settings take affect without a controller restart. I don't see anywhere where if they update the parallelism value, that the new number will take affect.

Reply:
On config change controller notifies throttler using `throttler.SetParallelism` and throttler triggers required number of workflows .

```
func (t *throttler) SetParallelism(parallelism *int) {
	t.throttleLock.Lock()
	defer t.throttleLock.Unlock()
	t.parallelism = parallelism
	t.queueThrottled()
}
```

You might also check `TestSetParallelismWithPriority` unit test